### PR TITLE
new-feature:add k3portlock

### PR DESCRIPTION
**Pull request by IloveKanade**
The net portlock has been modified according to the specification

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+_extends: gh-config
+
+repository:
+  description: Portlock is a cross-process lock that is implemented with `tcp` port binding.
```
Comment (drmingdrmer) [inline .github/settings.yml]: The package name now is `k3protlock`:D

File: README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# k3portlock
+
+[![Action-CI](https://github.com/pykit3/k3portlock/actions/workflows/python-package.yml/badge.svg)](https://github.com/pykit3/k3portlock/actions/workflows/python-package.yml)
+[![Build Status](https://travis-ci.com/pykit3/k3portlock.svg?branch=master)](https://travis-ci.com/pykit3/k3portlock)
+[![Documentation Status](https://readthedocs.org/projects/k3portlock/badge/?version=stable)](https://k3portlock.readthedocs.io/en/stable/?badge=stable)
+[![Package](https://img.shields.io/pypi/pyversions/k3portlock)](https://pypi.org/project/k3portlock)
+
+Portlock is a cross-process lock that is implemented with `tcp` port binding.
```
Comment (drmingdrmer) [inline README.md]: And all the `Portlock` should be replaced.

File: portlock.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        lck = key
+        print('lock is', i, lck)
+        l = Portlock(lck, timeout=8)
+        r = l.try_lock()
+        if not r:
+            print('collide', i, l.addr)
+            print(l.socks)
+
+        dd[l.addr] = i
+        ls.append(l)
```
Comment (drmingdrmer) [inline portlock.py]: Is it possible moving this test to `test/` directory as a common unit test?

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
 -r _building/requirements.txt
 
 
-k3ut~=0.1.7
+k3ut~=0.1.15
```
Comment (drmingdrmer) [inline requirements.txt]:
```suggestion
k3ut>=0.1.15,<0.2
```

With `~=`, a minor version change of `k3ut` results in a version conflict, if two package depends on different versions of `k3ut`.

`~=0.x` is compatible with `0.x.1`, `0.x.2`. While `~=0.x.y` is compatible only with `0.x.y`.

**Review by drmingdrmer — COMMENTED** [review]
Some term should be updated, since the package name changes.

File: .github/settings.yml (hunk 1 above)
Comment (IloveKanade) [reply .github/settings.yml]: all down~

File: requirements.txt (hunk 4 above)
Comment (IloveKanade) [reply requirements.txt]: all down~

File: test/test_k3portlock.py [hunk 5]
```diff
@@ -128,3 +130,21 @@ def test_sleep_time(self):
...
+            lck = key
+            print('lock is', i, lck)
+            l = k3portlock.Portlock(lck, timeout=8)
+            r = l.try_lock()
+            if not r:
+                print('collide', i, l.addr)
+                print(l.socks)
+
+            dd[l.addr] = i
+            ls.append(l)
```
Comment (drmingdrmer) [inline test/test_k3portlock.py]: Hm...I mean, make it a real **test**, with assertion that the collision rate should be slower than some value etc.

**Review by drmingdrmer — APPROVED** [review]
LGTM

**Pull request merged**